### PR TITLE
Fix SEGV in transcribe_type when compiling libcore 1.29

### PR DESCRIPTION
**Pull request by CohenArthur**
Needs #1513, Skip reviewing the first commit

### CODE REVIEW

File: gcc/rust/ast/rust-ast.h [hunk 1]
```diff
@@ -1868,6 +1869,10 @@ class ASTFragment
    */
 
   bool is_single_fragment () const { return nodes.size () == 1; }
+  bool is_single_fragment (SingleASTNode::NodeType expected) const
```
Comment (philberty) [inline gcc/rust/ast/rust-ast.h:1872]: Not sure about the name of this method. Its a pet nit of mine that I prefer to use overloading only on methods where they already take arguments. It feels like a new method to check if is_single_fragment and that single fragment is of a particular type.

How about `is_single_fragment_of_kind (...)`

File: gcc/rust/ast/rust-ast.h [hunk 2]
```diff
@@ -1874,9 +1874,39 @@ class ASTFragment
     return is_single_fragment () && nodes[0].get_kind () == expected;
   }
 
-  bool is_single_fragment_kind (SingleASTNode::NodeType kind) const
+  void assert_single_fragment (SingleASTNode::NodeType expected) const
```
Comment (philberty) [inline gcc/rust/ast/rust-ast.h:1877]: ah ok so it gets renamed here anyway :)

**Review by philberty — APPROVED** [review]
LGTM

File: gcc/rust/ast/rust-ast.h (hunk 1 above)
Comment (CohenArthur) [reply gcc/rust/ast/rust-ast.h:1872]: Ah, I see what you mean. I'll rename it. `assert_single_fragment` kinda serves the same purpose but it also crashes if the assertion fails, so it's a bit different from `is_single_fragment`

**Comment by CohenArthur** [comment]
bors retry

**Pull request merged**